### PR TITLE
Updated str_contains to preg_match for better support on windows

### PR DESCRIPTION
**Pull request by percymamedy**
I have changed the str_contains to preg_match it does solve the problem on a windows system, could you test if it still works on other environments. Thanks

### CODE REVIEW

File: src/Scopes/QueryTracer.php [hunk 1]
```diff
@@ -28,7 +28,7 @@ public function apply(Builder $builder, Model $model)
 
         foreach ($traces as $trace) {
             // Find the first non-vendor-dir file in the backtrace
-            if (isset($trace['file']) && ! str_contains($trace['file'], '/vendor/')) {
+            if (isset($trace['file']) && ! preg_match('/vendor/', $trace['file'])) {
```
Comment (fitztrev) [inline src/Scopes/QueryTracer.php]: This change would just check the path for `vendor` instead of `/vendor/`.

**Comment by percymamedy** [comment]
I see your point. I have added back str_contains but with another test for windows when the separators are `\vendor\` intead of `/vendor/`. Tell me what you think

**Comment by fitztrev** [comment]
What file path is being returned when you try to use this package as-is? If your file path contains `\vendor\`, using `DIRECTORY_SEPARATOR` might be the way to go.

**Comment by percymamedy** [comment]
I am on a windows system and as I pointed out in this [Issue](https://github.com/fitztrev/query-tracer/issues/1) when I use the package as it is, all my queries are shown as being executed from `"vendor\laravel\framework\src\Illuminate\Database\Eloquent\Builder.php" and "query.line" <> "1124"` because on windows directory separators are `\` instead of `/`

Currently the package checks for `/vendor/` when performing its backtrace which does not work on windows system.

**Comment by fitztrev** [comment]
Ok. If you want to update this and just change the original `str_contains` function to check for `'vendor' . DIRECTORY_SEPARATOR`, that should fix it and I'll merge it in.

**Comment by percymamedy** [comment]
OK I have change the original str_contains function to check for `DIRECTORY_SEPARATOR.'vendor' .DIRECTORY_SEPARATOR`

**Comment by fitztrev** [comment]
Ok, thanks. This fixes it when you run it in your application even though the path you shared starts with `vendor\` instead of `\vendor\`?

**Comment by percymamedy** [comment]
I actually did not share the full path because it was just to give and idea of which file was being shown  by the package the full path was actually `C:\xampp\htdocs\repos\my-project\vendor\laravel\framework\src\Illuminate\Database\Eloquent\Builder.php`

But yes using `DIRECTORY_SEPARATOR` does fix the problem on windows.

**Comment by fitztrev** [comment]
Thanks!

**Comment by percymamedy** [comment]
Thanks to you...:)

**Pull request merged**